.github/workflows/ci.yml

```
@@ -173,6 +173,15 @@ jobs:
               ../build/examples/trixi_controller_data_f . libelixir_p4est2d_dgsem_euler_sedov.jl
         env:
           LIBTRIXI_DEBUG: all
+      - name: Run julia tests with coverage
+        if: ${{ matrix.test_type == 'coverage' }}
+        uses: julia-actions/julia-runtest@v1
+        with:
+          coverage: true
+          project: ./LibTrixi.jl
+        env:
+          JULIA_DEPOT_PATH: ~/.julia
+          LIBTRIXI_DEBUG: all
```

```suggestion
      - name: Run julia tests with coverage
        uses: julia-actions/julia-runtest@v1
        with:
          coverage: ${{ matrix.test_type == 'coverage' }}
          project: ./LibTrixi.jl
        env:
          JULIA_DEPOT_PATH: ~/.julia
          LIBTRIXI_DEBUG: all
```
I wonder if this works, and if yes, whether this might make more sense: Always run the tests, but only run with coverage reporting if it is enabled globally?

---

Nice idea! Will try.

---

It seems to work 👍
/r

---

Seems to work. Great!

Just something I noticed: All packages are precompiled again in the Julia testing step. There is no way getting around this because the test happens in a completely different Julia project, right?

---

> There is no way getting around this because the test happens in a completely different Julia project, right?

Hm. No, not necessarily. I wonder whether it's got to do with the `--coverage` settings, i.e., if it needs to be recompiled if coverage is enabled? Maybe you can ask this on Discourse and CC me, would be interesting to hear what the community has to say...